### PR TITLE
Add Application Protocols and coreJSON under FreeRTOS-Plus/Source

### PR DESCRIPTION
**Pull request by yourslab**
Add Application Protocols and coreJSON under FreeRTOS-Plus/Source

Description
-----------

This adds coreMQTT, coreJSON, and Device Shadow as submodules to the following directories:
coreMQTT: `FreeRTOS-Plus/Source/Application-Protocols/coreMQTT`
coreJSON: `FreeRTOS-Plus/Source/coreJSON`
Platform files: `FreeRTOS-Plus/Source/Application-Protocols/platform` (Moved from `FreeRTOS-Plus\Source\FreeRTOS-IoT-Libraries-LTS-Beta2\c_sdk\platform` in `lts-development`)
Device Shadow: `FreeRTOS/FreeRTOS-Plus/Source/AWS/device-shadow-for-aws-iot-embedded-sdk`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by muneebahmed10 — APPROVED** [review]
Is JSON a protocol?

File: .gitmodules [hunk 1]
```diff
@@ -14,5 +14,5 @@
 	path = FreeRTOS-Plus/Source/Application-Protocols/coreMQTT
 	url = https://github.com/FreeRTOS/coreMQTT
 [submodule "FreeRTOS-Plus/Source/Application-Protocols/coreJSON"]
```
Comment (muneebahmed10) [inline .gitmodules:16]: Does this not need to be changed as well?
```suggestion
[submodule "FreeRTOS-Plus/Source/coreJSON"]
```

File: FreeRTOS-Plus/Source/Application-Protocols/platform/freertos/mbedtls/mbedtls_freertos_port.c [hunk 2]
```diff
@@ -0,0 +1,286 @@
+/*
```
Comment (muneebahmed10) [inline FreeRTOS-Plus/Source/Application-Protocols/platform/freertos/mbedtls/mbedtls_freertos_port.c:1]: Were these platform files meant to be in this PR? Seems like an unrelated change and not part of the PR description

**Review by muneebahmed10 — APPROVED** [review]

File: FreeRTOS-Plus/Source/Application-Protocols/platform/freertos/mbedtls/mbedtls_freertos_port.c (hunk 2 above)
Comment (yourslab) [reply FreeRTOS-Plus/Source/Application-Protocols/platform/freertos/mbedtls/mbedtls_freertos_port.c:1]: I've updated the PR description

**Review by abhidixi11 — APPROVED** [review]

**Review by n9wxu — APPROVED** [review]
